catalyst-gateway/bin/src/service/api/cardano/staking/assets_get.rs

```diff
@@ -185,12 +184,18 @@ async fn get_txo_by_txn(
         let entry = txo_info
             .assets
             .entry(row.policy_id.clone())
-            .or_insert(TxoAssetInfo {
-                id: row.policy_id,
-                name: row.policy_name,
-                amount: num_bigint::BigInt::ZERO,
-            });
-        entry.amount += row.value;
+            .or_insert_with(Vec::new);
+
+        match entry.iter_mut().find(|item| item.id == row.policy_id) {
+            Some(item) => item.amount += row.value.to_i128().unwrap_or(0),
```

Comment:
Zero is an invalid value.
This should fail to add the row, and log an error if it is unconvertable.
It should not fail to generate a result, this row should not be included.